src/main/java/com/google/api/codegen/util/Name.java

```diff
@@ -136,6 +136,7 @@ public String toLowerUnderscore() {
   private String toUnderscore(CaseFormat caseFormat) {
     List<String> newPieces = new ArrayList<>();
     for (NamePiece namePiece : namePieces) {
+      namePiece = replaceAcronyms(namePiece);
```

Comment:
I think it might make more sense to do this in Name.upperCamel; it is the entry point for upper camel strings.